go.mod

```diff
@@ -145,7 +145,7 @@ require (
 	github.com/siderolabs/grpc-proxy v0.4.0
 	github.com/siderolabs/kms-client v0.1.0
 	github.com/siderolabs/net v0.4.0
-	github.com/siderolabs/siderolink v0.3.6
```

Comment:
can we please remove this unrelated changes (?), as this should be a backportable PR, so just the fix.

Reply:
We can but it will fail on `e2e-siderolink` since `github.com/siderolabs/siderolink v0.3.6` is buggy on the server side. But it doesn't affect client part. Reverted those changes.

Reply:
if that is the case, let's do a separate PR and mark it to be backpored, but it's hard to manage two fixes in one PR